### PR TITLE
JAMES-2289 MailQueueFactories should return a list of created queues

### PR DESCRIPTION
**Pull request by chibenwa**

### CODE REVIEW

File: server/protocols/jmap/src/test/java/org/apache/james/jmap/send/MailSpoolTest.java [hunk 1]
```diff
@@ -43,7 +46,27 @@
...
-        mailSpool = new MailSpool(name -> myQueue);
+        mailSpool = new MailSpool(new SingletonMailQueueFactory(myQueue));
+    }
+
+    private static class SingletonMailQueueFactory implements MailQueueFactory {
+
+        private final MailQueue mailQueue;
+
+        public SingletonMailQueueFactory(MailQueue mailQueue) {
+
```
Comment (rouazana) [inline server/protocols/jmap/src/test/java/org/apache/james/jmap/send/MailSpoolTest.java]: blank line to remove

File: server/protocols/jmap/src/test/java/org/apache/james/jmap/send/MailSpoolTest.java [hunk 2]
```diff
@@ -43,7 +46,27 @@
     public void setup() {
         myQueue = new MemoryMailQueueFactory.MemoryMailQueue(MailQueueFactory.SPOOL, new RawMailQueueItemDecoratorFactory());
 
-        mailSpool = new MailSpool(name -> myQueue);
+        mailSpool = new MailSpool(new SingletonMailQueueFactory(myQueue));
+    }
+
+    private static class SingletonMailQueueFactory implements MailQueueFactory {
```
Comment (chibenwa) [inline server/protocols/jmap/src/test/java/org/apache/james/jmap/send/MailSpoolTest.java]: Should directly use MemoryMailQueueFactory.

**Comment by chibenwa** [comment]
test this please

```
[db5efe1332d398fa31edbc2ba5b2e8685ddb63f6] Tests in error: 
[db5efe1332d398fa31edbc2ba5b2e8685ddb63f6]   DefaultMailboxesProvisioningFilterTest.createMailboxesIfNeededShouldNotGenerateExceptionsInConcurrentEnvironment:83 » Execution
```

**Comment by chibenwa** [comment]
merged

**Pull request closed without merging**